spec/services/filter_application_choices_for_providers_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe FilterApplicationChoicesForProviders do
+  describe '.call' do
+    let(:application_choices) do
+      create_list(:application_choice, 2)
+      ApplicationChoice.all
```

Comment:
Do we need this line? By the looks of it, this could be a one line `let`

Reply:
We do need it yes, it's a side effect of the described class expecting all the joins or includes to be made before filtering.
`create_list` returns an array which won't respond to the `.joins` statements necessary in the tests.

Reply:
Ah yes, probably not one for this PR then. Maybe worth a tech debt card?

Reply:
:+1: Will add a tech debt story